gpii/node_modules/contextManager/src/CASConnector.js

```diff
@@ -0,0 +1,136 @@
+/*
+* Context Manager
+*
+* Copyright 2015 Raising the Floor - International
+*
+* Licensed under the New BSD license. You may not use this file except in
+* compliance with this License.
+*
+* The research leading to these results has received funding from the European Union's
+* Seventh Framework Programme (FP7/2007-2013)
+* under grant agreement no. 289016.
+*
+* You may obtain a copy of the License at
+* https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+(function () {
+    "use strict";
+
+    var fluid = require("infusion"),
+        io = require("socket.io-client"),
+        gpii = fluid.registerNamespace("gpii");
+
+    fluid.registerNamespace("gpii.contextManager.CASConnector");
+
+    fluid.defaults("gpii.contextManager.CASConnector", {
+        gradeNames: ["fluid.eventedComponent", "autoInit"],
+        members: {
+            socket: null,
+            CASUrl: "http://localhost:8888/context-stream"
+        },
+        events: {
+            onConnectRequest: null,
+            onEmitRequest: null
+        },
+        listeners: {
+            onCreate: "gpii.contextManager.CASConnector.connectSocket",
+            onDestroy: "gpii.contextManager.CASConnector.disconnectSocket"
+        },
+        invokers: {
+            connectSocket: {
+                funcName: "gpii.contextManager.CASConnector.connectSocket",
+                args: [ "{that}", "{contextManager}" ]
+            },
+            handleInput: {
+                funcName: "gpii.contextManager.CASConnector.handleInput",
+                args: [ "{contextManager}", "{arguments}.0" ]
+            }
+        }
+    });
+
+    gpii.contextManager.CASConnector.disconnectSocket = function (that) {
+        if (that.socket) {
+            that.socket.disconnect();
+        };
+    }
+    gpii.contextManager.CASConnector.connectSocket = function (that) {
+        var location =  [2.194909,41.402455];
+        console.log("Connecting to socket at "+that.CASUrl);
+        that.socket = io.connect(that.CASUrl);
+
+        that.socket.on("connect", function () {
+             fluid.log("Connected to Context Aware Server");
+
+            that.socket.emit('location', location);
+            fluid.log("Location emitted to Contet Aware Server: "+location);
+
+            that.socket.on('new-user', function (id) {
+                console.log("Got user ID from Context Aware Server: " +id)
+                that.socket.emit('subscribe', id);
+                console.log("Subscribed to Context Aware Server")
+                that.socket.on('fired', function (data) {
+                   // console.log("Retrieved data from Context Aware Server: " + JSON.stringify(i));
+                   that.handleInput(data);
+                });
+            });
+        });
+        that.socket.on("error", function (a) {
+            console.log("Error: " +JSON.stringify(a));
+        })
```

Comment:
Hey @kaspermarkus, 

as we talked last Friday, we are accumulating subscriptions to the CAS (which is not the end of the world), but might become a problem for us during the review. As a quick fix, I suggest you to listen to "disconnect" events and close/open a new socket connection by hand.

To do that, you can use:

```
that.socket.on("disconnect", function (a) {
    that.socket.removeAllListeners();
    that.connectSocket();
}
```